### ISSUE TITLE
Fine tune to predict heart parmeter ?

### ISSUE THREAD

**Issue by dineshBhatotia** [harryjdavies/HeartGPT#2]
Hi @harryjdavies , can it be fine tuned to identify a heart parameter ?
In that data set there is one heart parameter for every 2 seconds, so it will not have length exactly equal to the ppg signal data points. Can you share some pointers if this pre-trained model can be fine-tuned in that case or do we need to create a separate pre-train model for that ?

Anyways this repo is actually really helpful to start with, thanks for making it open source :)

**Comment by harryjdavies** [comment]
Hi @dineshBhatotia , thank you for your question! I think one hopefully comprable example is that we fine tuned it for beat detection, where the training labels were 0 if there was no beat, and 1 at the position of a beat - this was a way of making the labels the same size as the input data. This was simple though as it required binary cross entropy.
If there is a set 2 seconds of data, you don't have to use the full block size. E.g you could fine-tune it by inputting just 100 samples (corresponding to 2 seconds) and just predicting one value. You could probably do the same if it was a parameter per each beat by just cropping beats and feeding each individual beat in. We have not experimented with this though. Best of luck and hope this was helpful

**Comment by dineshBhatotia** [comment]
Thanks for responding @harryjdavies, this is helpful. So if we use just 2 second data how can we map it to a single value ? Is it possible with current model architecture or do we need to train that from scratch. We do not have a large data set to create a GPT type model from scratch.

**Comment by harryjdavies** [comment]
@dineshBhatotia  I believe it would be possible from the current architecture. The full block size of 500 is set up to accomodate 10 seconds of PPG. You can instead just use the last 100 samples. As we did in the fine-tuning for AF, you can then set a new LM head that converts from model dimension (64) to single dimension (1), and then just take the last value outputted as your output (this is the token that will have had access to all other tokens in your input, due to the masked structure).

**Comment by dineshBhatotia** [comment]
Sure, will try it out @harryjdavies. Thank you!
